x-pack/legacy/plugins/apm/public/components/app/ServiceMap/useRefHeight.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { MutableRefObject, useRef } from 'react';
+import { useWindowSize } from 'react-use';
+
+export function useRefHeight(): [
+  MutableRefObject<HTMLDivElement | null>,
+  number
+] {
+  const ref = useRef<HTMLDivElement>(null);
+  const windowHeight = useWindowSize().height;
+  const approximateGuessOfDefaultTopOffset = 230;
+  const topOffset =
+    ref.current?.getBoundingClientRect()?.top ??
+    approximateGuessOfDefaultTopOffset;
```

Comment:
does it make a noticeable difference if there is no default value here? If `height` defaults to `undefined` or `null`, then gets set as a prop for the `Cytoscape` component, would it just render with the default height anyways?

Reply:
It doesn't make much of a difference, but on the initial load the height is the full window height, so there's a scrollbar there until we get the ref and the top offset.

I'm changing this so we just return 0 for the height if we don't yet have the offset.

Reply:
Updated in f9a1448011.